src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -114,13 +118,33 @@ protected DurableTaskStepDescriptor() {
             return ws;
         }
 
+        private @Nonnull PrintStream logger() {
+            TaskListener l = listener;
+            if (l == null) {
+                StepContext context = getContext();
+                try {
+                    l = context.get(TaskListener.class);
+                    if (l != null) {
+                        LOGGER.log(Level.WARNING, "JENKINS-34021: DurableTaskStep.Execution.listener not restored in {0}", context);
+                    } else {
+                        LOGGER.log(Level.WARNING, "JENKINS-34021: TaskListener not even available upon request in {0}", context);
+                        l = new LogTaskListener(LOGGER, Level.FINE);
```

Comment:
Well, if this ever happens again, a dev is going to need to supply user instructions on enabling logging etc.